forch/forchestrator.py

```diff
@@ -414,32 +414,26 @@ def _register_handlers(self):
                 event.timestamp, event.dp_name, event.port, event.state)),
             (FaucetEvent.StackTopoChange, fcoll.process_stack_topo_change_event),
             (FaucetEvent.PortChange, fcoll.process_port_change),
+            (FaucetEvent.PortChange, self._device_report_server_process_port_change),
             (FaucetEvent.L2Learn, lambda event: fcoll.process_port_learn(
                 event.timestamp, event.dp_name, event.port_no, event.eth_src, event.l3_src_ip)),
+            (FaucetEvent.L2Learn, self._device_report_server_process_port_learn),
             (FaucetEvent.L2Expire, lambda event: fcoll.process_port_expire(
                 event.timestamp, event.dp_name, event.port_no, event.eth_src, event.vid)),
-            *self._get_device_report_server_event_handlers()
         ]
 
         self._faucet_events.register_handlers(handlers)
 
-    def _get_device_report_server_event_handlers(self):
-        if not self._device_report_server:
-            return []
-
-        return [
-            (
-                FaucetEvent.PortChange,
-                lambda event: self._device_report_server.process_port_change(
-                    event.timestamp, event.dp_name, event.port_no,
-                    event.status and event.reason != 'DELETE')
-            ),
-            (
-                FaucetEvent.L2Learn,
-                lambda event: self._device_report_server.process_port_learn(
-                    event.dp_name, event.port_no, event.eth_src)
-            ),
-        ]
+    def _device_report_server_process_port_change(self, event):
```

Comment:
nit: personally I don't believe it's good to over-specify the function name, since really it could just be "process_port_change" -- since if, for some reason there's another thing you want to do on a port change then you could also use this function.  Really there's no need to have two entries in the dispatch table (you could move the fcoll.process_port_change call into this function).  Doesn't really matter -- thus the nit.  Change or not as you see fit.